Core add-ons are missing

### Describe the issue you are experiencing

Core add-ons are missing except from those that were installed when home assisstant itself was installed.
Icons of those add-ons are missing too.

When manually checking add-ons for updates, errors are logged in supervisor log.

This issue is a blocker for me, because there is no way for me to install mosquitto broker and given that I can't use Zigbee2mqtt.

Core add-ons that are installed don't get updated.

### What is the used version of the Supervisor?

supervisor-2022.05.2

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

5.15.38

### What version of Home Assistant Core is installed?

core-2022.5.5

### Steps to reproduce the issue

1. Go to Settings > Add-ons;
2. Click "Add-on store" button bottom right of the screen;
3. Click three dots top right of the screen;
4. Select "Check for updates" from drop list;
5. Check Supervisor log for errors.
![Screenshot 2022-05-19 at 17 40 16](https://user-images.githubusercontent.com/47085307/169323283-c33266c1-891b-43b0-8b34-426504d9b01c.png)

### Anything in the Supervisor logs that might be useful for us?

```txt
Full logs
AssertionError: Remote 'origin' has no refspec set.
You can set it as follows: 'git config --add "remote.origin.fetch +refs/heads/*:refs/heads/*"'.
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/repository.py", line 110, in update
    await self.git.pull()
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 112, in wrapper
    raise JobException() from err
supervisor.exceptions.JobException
22-05-19 15:46:30 INFO (MainThread) [supervisor.store] Loading add-ons from store: 42 all - 0 new - 0 remove
22-05-19 15:46:30 INFO (MainThread) [supervisor.store] Loading add-ons from store: 42 all - 0 new - 0 remove
22-05-19 16:04:45 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-05-19 16:20:09 INFO (MainThread) [supervisor.host.info] Updating local host information
22-05-19 16:20:10 INFO (MainThread) [supervisor.host.services] Updating service information
22-05-19 16:20:10 INFO (MainThread) [supervisor.host.network] Updating local network information
22-05-19 16:20:10 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
22-05-19 16:20:10 INFO (MainThread) [supervisor.host.manager] Host information reload completed
22-05-19 16:35:29 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.SUPERVISOR
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_FAILED/ContextType.DNS_SERVER
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_IPV6_ERROR/ContextType.DNS_SERVER
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.check] System checks complete
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
22-05-19 16:44:32 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
22-05-19 17:06:16 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-05-19 17:37:04 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-05-19 17:41:35 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository
22-05-19 17:41:35 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository
22-05-19 17:41:35 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/esphome/home-assistant-addon repository
22-05-19 17:41:35 ERROR (MainThread) [supervisor.jobs] Unhandled exception: Remote 'origin' has no refspec set.
You can set it as follows: 'git config --add "remote.origin.fetch +refs/heads/*:refs/heads/*"'.
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/store/git.py", line 155, in pull
    await self.sys_run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 876, in fetch
    self._assert_refspec()
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 835, in _assert_refspec
    raise AssertionError(msg % (self.name, self.name))
AssertionError: Remote 'origin' has no refspec set.
You can set it as follows: 'git config --add "remote.origin.fetch +refs/heads/*:refs/heads/*"'.
22-05-19 17:41:39 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-102028' coro=<Repository.update() done, defined at /usr/src/supervisor/supervisor/store/repository.py:106> exception=JobException()>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 106, in wrapper
    return await self._method(*args, **kwargs)
  File "/usr/src/supervisor/supervisor/store/git.py", line 155, in pull
    await self.sys_run_in_executor(
  File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 876, in fetch
    self._assert_refspec()
  File "/usr/local/lib/python3.9/site-packages/git/remote.py", line 835, in _assert_refspec
    raise AssertionError(msg % (self.name, self.name))
AssertionError: Remote 'origin' has no refspec set.
You can set it as follows: 'git config --add "remote.origin.fetch +refs/heads/*:refs/heads/*"'.
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/store/repository.py", line 110, in update
    await self.git.pull()
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 112, in wrapper
    raise JobException() from err
supervisor.exceptions.JobException
22-05-19 17:41:39 INFO (MainThread) [supervisor.store] Loading add-ons from store: 42 all - 0 new - 0 remove
22-05-19 17:41:39 INFO (MainThread) [supervisor.store] Loading add-ons from store: 42 all - 0 new - 0 remove
```


### Additional information

I saw other HA users having this very problem, but there was very little if no effort form HA developers to try and address this issue.

Would be nice to finally get proper help and the solution.

--- a/supervisor/store/git.py
+++ b/supervisor/store/git.py
@@ -184,6 +184,7 @@ class GitRepo(CoreSysAttributes):
                 git.NoSuchPathError,
                 git.GitCommandError,
                 ValueError,
+                AssertionError,
             ) as err:
                 _LOGGER.error("Can't update %s repo: %s.", self.url, err)
                 self.sys_resolution.create_issue(

